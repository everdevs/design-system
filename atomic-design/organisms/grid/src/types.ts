import React from "react";
import { GridConfig, PropsWithTheme } from "@evernest/theme";

export type ColSpanValue = string | number;

export interface ColSpan {
	xs?: ColSpanValue;
	s?: ColSpanValue;
	m?: ColSpanValue;
	l?: ColSpanValue;
}

export interface ColumnProps extends ColSpan, PropsWithTheme {
	raw?: boolean;
	flex?: boolean;
	order?: number;
}
export type ColumnType = React.ReactElement<ColumnProps> | React.ReactElement<ColumnProps>[];

export enum Alignment {
	start = "flex-start",
	end = "flex-end",
	center = "center",
}

export interface RowProps extends PropsWithTheme {
	children?: ColumnType;
	raw?: boolean;
	noWrap?: boolean;
	reverse?: boolean;
	justify?: Alignment;
}

export type RowType = React.ReactElement<RowProps> | React.ReactElement<RowProps>[];

export interface GridProps extends PropsWithTheme {
	children?: RowType;
	overflow?: boolean;
}

export type GridOverlayPosition = "fixed" | "absolute";

export interface GridOverlayGridProps extends GridProps {
	position?: GridOverlayPosition;
}

export interface GridOverlayProps extends GridOverlayGridProps {
	initialActive?: boolean;
	toggle?: boolean;
}

export interface ViewportSize {
	m: boolean;
	mu: boolean;
	l: boolean;
	lu: boolean;
	xl: boolean;
}

export interface GridContextConfig extends GridConfig {
	viewport: ViewportSize;
}

export interface GridContextProps {
	grid: GridConfig;
}

export interface BoxProps {
	removePadding?: boolean;
	addPadding?: boolean;
	removeGutter?: boolean;
	addGutter?: boolean;
	flex?: boolean;
}
