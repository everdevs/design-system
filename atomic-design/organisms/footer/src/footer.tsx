import { css } from "@emotion/core";
import styled from "@emotion/styled";
import React from "react";
import { StyledFooterProps, FooterProps } from "./types";

export const StyledFooter = styled.div<StyledFooterProps>`
	padding: var(--spacing-xl) 0;
	${({ theme: { palette } }) => css`
		background: ${palette.darkGrey.css};
		color: ${palette.white.css};
	`};
`;

export const Footer = React.forwardRef<FooterProps, StyledFooterProps>((props, ref) => (
	<StyledFooter {...props} ref={ref} />
));
