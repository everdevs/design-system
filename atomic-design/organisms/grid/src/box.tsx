import { css } from "@emotion/react";
import styled from "@emotion/styled";
import React from "react";
import { GUTTER, PADDING } from "./constants";
import { BoxProps } from "./types";

const StyledBox = styled.div<BoxProps>`
	flex: 1;
	${({ flex }) => css`
		display: ${flex ? "flex" : "block"};
	`};
	${({ removePadding, removeGutter, addGutter, addPadding }) => {
		if (removePadding) {
			if (removeGutter) {
				return css`
					margin-left: calc((var(${PADDING}) + var(${GUTTER})) * -1px);
					margin-right: calc((var(${PADDING}) + var(${GUTTER})) * -1px);
				`;
			}
			return css`
				margin-left: calc(var(${PADDING}) * -1px);
				margin-right: calc(var(${PADDING}) * -1px);
			`;
		}
		if (removeGutter) {
			return css`
				margin-left: calc(var(${GUTTER}) * -1px);
				margin-right: calc(var(${GUTTER}) * -1px);
			`;
		}
		if (addGutter) {
			if (addPadding) {
				return css`
					padding-left: calc(var(${PADDING}) * 1px);
					padding-right: calc(var(${PADDING}) * 1px);
					margin-left: calc(var(${GUTTER}) * 1px);
					margin-right: calc(var(${GUTTER}) * 1px);
				`;
			}
			return css`
				margin-left: calc(var(${GUTTER}) * 1px);
				margin-right: calc(var(${GUTTER}) * 1px);
			`;
		}
		return css`
			padding-left: calc(var(${PADDING}) * 1px);
			padding-right: calc(var(${PADDING}) * 1px);
		`;
	}};
`;

export const Box: React.FC<BoxProps> = props => {
	return <StyledBox {...props} />;
};
