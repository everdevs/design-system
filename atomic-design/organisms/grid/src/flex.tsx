import styled from "@emotion/styled";

export const Flex = styled.div`
	display: flex;
	justify-content: space-between;
	align-content: center;
	align-items: center;
`;
