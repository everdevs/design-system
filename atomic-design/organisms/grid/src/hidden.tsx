import styled from "@emotion/styled";
import { PropsWithTheme } from "@evernest/theme";
import { css } from "@emotion/react";

export interface HiddenProps extends PropsWithTheme {
	display?: string;
	s?: boolean;
	m?: boolean;
	l?: boolean;
}
export const Hidden = styled.div<HiddenProps>`
	${({ theme: { mq }, display, s, m, l }) => css`
		display: ${s ? display : "none"};
		@media ${mq.m} {
			display: ${m ? display : "none"};
		}

		@media ${mq.l} {
			display: ${l ? display : "none"};
		}
	`};
`;

Hidden.defaultProps = {
	display: "block",
};
