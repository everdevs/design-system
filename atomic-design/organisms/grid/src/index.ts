export * from "./box";
export * from "./column";
export * from "./constants";
export * from "./flex";
export * from "./hidden";
export * from "./grid";
export * from "./grid-overlay";
export * from "./row";
export * from "./stage";
export * from "./types";
