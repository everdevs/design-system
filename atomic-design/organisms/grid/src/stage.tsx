import { COLCOUNT, COLSPAN } from "./constants";
import styled from "@emotion/styled";

export const Stage = styled.div`
	${COLCOUNT}: var(${COLSPAN});
	display: flex;
	width: 100vw;
	margin-left: calc((-100vw + 100%) / 2);
	margin-right: calc((-100vw + 100%) / 2);
`;
