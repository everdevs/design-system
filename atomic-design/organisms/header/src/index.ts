export * from "./header";
export * from "./types";
export * from "./utils";
