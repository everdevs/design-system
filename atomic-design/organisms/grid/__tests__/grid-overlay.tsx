import React from "react";
import { mount } from "enzyme";
import { GridOverlay } from "../src";
import { createTree, Wrapper } from "@evernest/dev-helpers";

test("GridOverlay can be initially active", () => {
	const tree = createTree(
		<Wrapper>
			<GridOverlay initialActive />
		</Wrapper>
	);
	expect(tree).toMatchSnapshot();
});

test("GridOverlay can have position absolute", () => {
	const tree = createTree(
		<Wrapper>
			<GridOverlay position="absolute" />
		</Wrapper>
	);
	expect(tree).toMatchSnapshot();
});

test("GridOverlay can have a toggle button", () => {
	const tree = createTree(
		<Wrapper>
			<GridOverlay toggle />
		</Wrapper>
	);
	expect(tree).toMatchSnapshot();
	const wrapper = mount(
		<Wrapper>
			<GridOverlay toggle />
		</Wrapper>
	);
	wrapper.contains("[data-test-id='grid-toggle']");
});

test("GridOverlay can be toggled", () => {
	const wrapper = mount(
		<Wrapper>
			<GridOverlay toggle />
		</Wrapper>
	);
	const toggle = wrapper.find("[data-test-id='grid-toggle']").at(0);
	expect(toggle.text()).toEqual("grid");
	toggle.simulate("click");
	expect(toggle.text()).toEqual("gridOff");
});
