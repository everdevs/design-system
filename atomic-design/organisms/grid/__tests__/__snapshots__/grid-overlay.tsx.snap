// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridOverlay can be initially active 1`] = `
<div
  className="css-qmizr8"
>
  <div
    className="css-1jpbipk"
    position="fixed"
  >
    <div
      className="css-yrxebj"
      justify="flex-start"
    >
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
      <div
        className="css-13b9fqp"
        s={1}
        suppressHydrationWarning={true}
      >
         
      </div>
    </div>
  </div>
</div>
`;

exports[`GridOverlay can have a toggle button 1`] = `
<div
  className="css-qmizr8"
>
  <button
    className="css-vacqy9"
    data-test-id="grid-toggle"
    onClick={[Function]}
  >
    grid
  </button>
</div>
`;

exports[`GridOverlay can have position absolute 1`] = `
<div
  className="css-qmizr8"
/>
`;
